metadata-ingestion/source_docs/feast_repository.md

```diff
@@ -0,0 +1,52 @@
+# Feast Repository
+
+For context on getting started with ingestion, check out our [metadata ingestion guide](../README.md).
+
+## Setup
+
+To install this plugin, run `pip install 'acryl-datahub[feast-repository]'`.
+
+## Capabilities
+
+This plugin extracts:
+
+- Entities as [`MLPrimaryKey`](https://datahubproject.io/docs/graphql/objects#mlprimarykey)
+- Features as [`MLFeature`](https://datahubproject.io/docs/graphql/objects#mlfeature)
+- Feature views and on-demand feature views as [`MLFeatureTable`](https://datahubproject.io/docs/graphql/objects#mlfeaturetable)
+- Batch and stream source details as [`Dataset`](https://datahubproject.io/docs/graphql/objects#dataset)
+
+## Quickstart recipe
+
+Check out the following recipe to get started with ingestion! See [below](#config-details) for full configuration options.
+
+For general pointers on writing and running a recipe, see our [main recipe guide](../README.md#recipes).
+
+```yaml
+source:
+  type: "feast-repository"
+  config:
+    # Coordinates
+    path: "/path/to/repository/"
+    # Options
+    environment: "PROD"
+
+sink:
+  # sink configs
+```
+
+## Config details
+
+Note that a `.` is used to denote nested fields in the YAML recipe.
+
+| Field         | Required | Default | Description                                |
+| ------------- | -------- | ------- | ------------------------------------------ |
+| `path`        | ✅       |         | Path to Feast repository.                  |
+| `environment` |          | `PROD`  | Environment to use when constructing URNs. |
```

Comment:
@danilopeixoto, does this source need any credentials to connect?

Reply:
Yes! But at Feast level. How does it work?

1. The team clones the Feast repository (eg. local, JupyterLab, or CI pipeline environment).
2. The team injects variables into the environment to use them in the Feast project definition:

```yaml
# feature_store.yaml

registry:
    path: 's3://${FEAST_S3_BUCKET_NAME}/metadata.db'
project: 'demo'
provider: 'local'
online_store:
    type: 'redis'
    redis_type: 'redis'
    connection_string: '${REDIS_HOSTNAME}:${REDIS_PORT},password=${REDIS_PASSWORD}'
offline_store:
    type: 'file'
flags:
    alpha_features: true
    direct_ingest_to_online_store: true
    on_demand_transforms: true
```

If the Feast CLI and SDK are working for the chosen offline/online/registry stores:

```console
feast apply
```

then DataHub source will also work:

```console
datahub ingest -c feast_to_datahub_recipe.yaml
```